### ISSUE TITLE
pls add marge armv7 and armv8

### ISSUE THREAD

**Issue by matrix3d** [Gamua/Adobe-Runtime-Support#242]

**Comment by marchbold** [comment]
Please describe your issue in detail including the version of AIR you are using, the platform and how to replicate.

**Comment by matrix3d** [comment]
All versions of the SDK are not supported and integrate the armv7 and armv8 versions. Now need to add a new feature, the same apk contains armv7 and armv8

**Comment by ajwfrost** [comment]
I think originally, Adobe had looked at packaging multiple ABIs into one APK, but then took this out because it would make an overly large APK file - and it's the APK that gets downloaded to an end user, so this means longer install times etc. Google/Android provided the "multiple APKs" option so that you can have multiple smaller APKs and the Play Store will determine which one to download onto an end user's device.

The ideal would be to support the Android App Bundle format but we are still facing issues here with having to package up different resources from different sources (ANEs) and trying to find a way to still have the programmatic access to the resources; the app bundle format only supports one resource block..

So: if you really wanted to, we could package ARMv7 and ARMv8 binaries together into one APK, but it's not recommended and I don't know what warnings you'd get from the Play Store when doing this!

**Comment by matrix3d** [comment]
The new format of the Google Store is great. However, there are many other stores or downloads directly from the website that do not support Android App Bundle.
Can you provide a function for developers to choose the libraries included in the apk, such as ```-arch armv7,armv8```,thanks

**Comment by ajwfrost** [comment]
Sounds like a good idea as a compromise! We'll see whether we can do just that i.e. -arch armv7,armv8 to include multiple ABIs into a single APK. I think this way you also wouldn't need to have the version number change between each architecture.

**Comment by matrix3d** [comment]
Hello, when will this feature be completed? Can you provide a simple tutorial to manually merge the apk of armv7 and armv8 when it is not completed now?

**Comment by KaseYu** [comment]
@ajwfrost Google/Android provided the "multiple APKs" It provides a way to coexist armv7 and armv8, but the Samsung Galaxy Tab A (SM-T515), which is part of the Android 10 system but only supports armv7, resulting in the app being The google play will show incompatibility and will not install.

**Comment by KaseYu** [comment]
@FliplineStudios Yes, that's how we set it up too, but for some Samsung devices, that's what happens when you can't install the above!
![Image 1](https://user-images.githubusercontent.com/13024120/84333287-e8bd9d80-abc1-11ea-8cb3-fde3cd57546b.png)

**Comment by FliplineStudios** [comment]
Hmm is there a reason why your 32-bit APK only supports up to API Level 27, and not further?  For ours we have our armv7 APK support 19+ and our armv8 support 21+.

Many Android devices are built on 64-bit hardware, but actually still use a 32-bit OS (including many Samsung tablets), so these will still require the older armv7 APKs and can't install armv8 APKs.  So since that tablet is using API 28 or 29 on a 32-bit OS, neither of your APKs would be compatible in this case.

**Comment by KaseYu** [comment]
It was set according to the shelf guide, so maybe we can change it for the next update!
![1591761694598](https://user-images.githubusercontent.com/13024120/84335576-bd3db180-abc7-11ea-9628-58dcdebfdaaa.jpg)
